bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/wrappers/veneer/TestBigtableVeneerApi.java

```diff
@@ -0,0 +1,154 @@
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase.wrappers.BigtableApi;
+import com.google.cloud.bigtable.hbase.wrappers.BigtableHBaseSettings;
+import com.google.cloud.bigtable.hbase.wrappers.classic.BigtableHBaseClassicSettings;
+import com.google.common.collect.Queues;
+import com.google.protobuf.Empty;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.apache.hadoop.conf.Configuration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class TestBigtableVeneerApi {
+
+  private static final String TEST_PROJECT_ID = "fake-project-id";
+  private static final String TEST_INSTANCE_ID = "fake-instance-id";
+  private static final String ROW_KEY = "test-row-key";
+  private static final String TABLE_ID = "test-table-id";
+  private Server server;
+  private FakeDataService fakeDataService = new FakeDataService();
+  private FakeAdminService fakeAdminService = new FakeAdminService();
+
+  private BigtableHBaseSettings bigtableHBaseSettings;
+  private BigtableApi bigtableApi;
+
+  @Before
+  public void setUp() throws IOException {
+    final int port;
+    try (ServerSocket s = new ServerSocket(0)) {
+      port = s.getLocalPort();
+    }
+    server =
+        ServerBuilder.forPort(port)
+            .addService(fakeDataService)
+            .addService(fakeAdminService)
+            .build();
+    server.start();
+
+    Configuration configuration = new Configuration(false);
+    configuration.set(BigtableOptionsFactory.PROJECT_ID_KEY, TEST_PROJECT_ID);
+    configuration.set(BigtableOptionsFactory.INSTANCE_ID_KEY, TEST_INSTANCE_ID);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY, "true");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_DATA_CHANNEL_COUNT_KEY, "1");
+    configuration.set(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY, "localhost:" + port);
+    configuration.set(BigtableOptionsFactory.BIGTABLE_USE_GCJ_CLIENT, "true");
+    bigtableHBaseSettings = BigtableHBaseClassicSettings.create(configuration);
+    bigtableApi = BigtableApi.create(bigtableHBaseSettings);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    bigtableApi.close();
+    if (server != null) {
+      server.shutdownNow();
+      server.awaitTermination();
+    }
+  }
+
+  @Test
+  public void testBigtableHBaseSettings() {
+    assertTrue(bigtableApi instanceof BigtableVeneerApi);
+    assertSame(bigtableHBaseSettings, bigtableApi.getBigtableHBaseSettings());
+  }
+
+  @Test
+  public void testAdminClient() throws Exception {
+    assertTrue(bigtableApi.getAdminClient() instanceof AdminClientVeneerApi);
```

Comment:
I think it's better call testGetAdminClient()

Reply:
Done